package/pyproject.toml

```diff
@@ -12,3 +12,61 @@ requires = [
     "setuptools",
     "wheel",
 ]
+
+[project]
+name = "MDAnalysis"
+version = 2.1.0-dev0
+license = {file = "LICENSE"}
+authors = [
+    {name = 'MDAnalysis Core Developers', email = 'mdanalysis@numfocus.org'}
+]
+maintainers = [
+    {name = 'MDAnalysis Core Developers', email = 'mdanalysis@numfocus.org'}
+]
+requires-python = ">=3.7"
+dependencies = [
```

Comment:
I wasn't too sure how to deal with the optional deps here.

Tests -> should I just add MDAnalysisTests as the dep? How do you dynamically set the version number in pyproject.toml?
Extra deps ->
  - Not sure what the keyword should be here (it's unclear if you can just say "extra").
  - How do we deal with conda only deps? Does it matter?

Reply:
`conda` only deps seems like a bad idea in general, not sure you can do much about them here..

As for `dependencies`, I think it should just be the *mandatory* runtime dependencies, and then `[project.optional-dependencies]` can contain sections for `test`, `doc`, and `dev` tools (I'd suggest just looking at i.e., the SciPy example `pyproject.toml` and adapting for our purposes, if you aren't already doing that).

Reply:
Sounds good, for now let's just skip `[project.optional-dependencies]` since most are conda based, I'll raise an issue on merge of this PR with all the things that can be added at a later date to augment the pyproject.toml.